File: app/src/main/java/ashunevich/uniconverter20/currencyapi/CurrencyContractor.java
package ashunevich.uniconverter20.currencyapi;

public interface CurrencyContractor {


    interface Presenter {
        void requestDataFromServer();
    }

    interface View {

        void onResponseFailure(Throwable throwable);

        void parseDataFromResponseToHashmap(CurrencyResponseObject object);
    }

    interface Interactor {

        interface OnFinishedListener {
            void onFinished(CurrencyResponseObject object);

            void onFailure(Throwable t);
        }

        void getCurrencyData(OnFinishedListener onFinishedListener);


    }
}

File: app/src/main/java/ashunevich/uniconverter20/currencyapi/CurrencyContractor.kt
package ashunevich.uniconverter20.currencyapi

open interface CurrencyContractor {
    open interface Presenter {
        fun requestDataFromServer()
    }

    open interface View {
        fun onResponseFailure(throwable: Throwable?)
        fun parseDataFromResponseToHashmap(`object`: CurrencyResponseObject?)
    }

    open interface Interactor {
        open interface OnFinishedListener {
            fun onFinished(`object`: CurrencyResponseObject?)
            fun onFailure(t: Throwable?)
        }

        fun getCurrencyData(onFinishedListener: OnFinishedListener)
    }
}